numpy/_core/src/multiarray/stringdtype/casts.cpp

```diff
@@ -1468,13 +1714,17 @@ string_to_timedelta(PyArrayMethod_Context *context, char *const data[],
         if (is_null) {
             if (has_null && !has_string_na) {
                 *out = NPY_DATETIME_NAT;
-                goto next_step;
+                in += in_stride;
+                out += out_stride;
+                continue;
             }
             s = *default_string;
         }
         if (is_nat_string(&s)) {
             *out = NPY_DATETIME_NAT;
-            goto next_step;
+            in += in_stride;
+            out += out_stride;
+            continue;
```

Comment:
if you're going to change this can you also do the same change in `stringdtype_ufuncs.cpp`? Or just leave it, I added the gotos because I was spending far too much time trying to understand the control flow and the repeated code was confusing me.

Reply:
Ahh, I should have left a note about this in the body of the PR. This change is because the compiler will not allow you to `goto` across variable definitions. So with cpp this is actually a compile time error.

I'll make the same changes in `stringdtype_ufuncs.cpp` as well, so that it matches.